### ISSUE TITLE
Fix S4275: FP when the value is encapsulated in another class

### ISSUE THREAD

**Issue by crypto-rsa** [SonarSource/sonar-dotnet#2867]
### Description

When the field represents a class which encapsulates the actual value returned from the property, S4275 is incorrectly raised.

### Repro steps

Consider this code:
```
public class Sample
{
    private class ValueWrapper<T>
    {
        public T Value { get; set; }
        // ...
    }

    private readonly ValueWrapper<double> _someMember = new ValueWrapper<double>();

    public double SomeMember
    {
        get => _someMember.Value;
        set => _someMember.Value = value;
    }
}
```

### Expected behavior

S4275 should not be raised because the setter _does_ access the corresponding field.

### Actual behavior

S4275 is raised on the setter.

### Related information

* SonarLint Version 4.14.0.12332
* Visual Studio Version 16.4.0

**Comment by andrei-epure-sonarsource** [comment]
thanks @crypto-rsa ! 
note for us: SLVS 4.14 embeds SonarC# 8.0

**Comment by pavel-mikula-sonarsource** [comment]
Same applies for VB.NET

```
Public Class Sample

    Public Class ValueWrapper(Of T)

        Public Property Value As T

    End Class

    Private ReadOnly _someMember As New ValueWrapper(Of Double)

    Public Property SomeMember As Double
        Get
            Return _someMember.Value
        End Get
        Set(value As Double)
            _someMember.Value = value
        End Set
    End Property

End Class
```
